pkg/pillar/types/global.go

```diff
@@ -19,6 +20,92 @@ const (
 	symlinkDir       = TmpDirname + "/GlobalConfig"
 )
 
+// ConfigItemStatus - Status of Config Items
+type ConfigItemStatus struct {
+	// Value - Current value of the item
+	Value string
+	// Err - Error from last config. nil if no error.
+	Err error
+}
+
+// GlobalStatus - Status of Global Conig Items.
+type GlobalStatus struct {
+	// ConfigItems - key : Item Key Str
+	ConfigItems map[string]ConfigItemStatus
+	// UnknownConfigItems - Unrecognized ConfigItems.
+	UnknownConfigItems map[string]ConfigItemStatus
+}
+
+func (gs *GlobalStatus) setItemValue(key, value string) {
+	item := gs.ConfigItems[key]
```

Comment:
This isn't very robust; will panic if the key isn't known, which might happen when we add more items over time.
item, ok := gs.ConfigItems[key]
if !ok {
  // add the item
}

Reply:
Hi Erik, I will add a check here.. But I think we should do log.Fatal here. There is really no use case for returning the error and making the calling code do the log.Fatalf. This should always succeed or crash.

Reply:
This is fine, isn't it? If element is found ( Error case ) - we will get the item. On normal case, we get default Item.. Which we then set the Value in and set it in map again. There is no need to check for error here..

It doesn't panic in this case - just returns the default value.. The following is the behavior:

"i := m["route"]
If the requested key doesn't exist, we get the value type's zero value. In this case the value type is int, so the zero value is 0:"

Reply:
Ah, didn't realize that's how the keys in maps work.